lib/ansible/modules/cloud/spotinst/spotinst_aws_elastigroup.py

```diff
@@ -0,0 +1,1509 @@
+              subnet_id: subnet-2b68a15c
+          image_id: ami-f173cc91
+          key_pair: spotinst-oregon
+          max_size: 15
+          min_size: 0
+          target: 0
+          unit: instance
+          monitoring: True
+          name: ansible-group
+          on_demand_instance_type: c3.large
+          product: Linux/UNIX
+          load_balancers:
+            - test-lb-1
+          security_group_ids:
+            - sg-8f4b8fe9
+          spot_instance_types:
+            - c3.large
+          do_not_update:
+            - image_id
+            - target
+      register: result
+    - debug: var=result
+
+
+# In this example we create a basic group configuration with a target tracking scaling policy defined
+
+- hosts: localhost
+  tasks:
+    - name: create elastigroup
+      spotinst_aws_elastigroup:
+          account_id: act-92d45673
+          state: present
+          risk: 100
+          availability_vs_cost: balanced
+          availability_zones:
+            - name: us-west-2a
+              subnet_id: subnet-79da021e
+          image_id: ami-f173cc91
+          fallback_to_od: true
+          tags:
+            - Creator: ValueOfCreatorTag
+            - Environment: ValueOfEnvironmentTag
+          key_pair: spotinst-labs-oregon
+          max_size: 10
+          min_size: 0
+          target: 2
+          unit: instance
+          monitoring: True
+          name: ansible-group-1
+          on_demand_instance_type: c3.large
+          product: Linux/UNIX
+          security_group_ids:
+            - sg-46cdc13d
+          spot_instance_types:
+            - c3.large
+          target_tracking_policies:
+            - policy_name: target-tracking-1
+              namespace: AWS/EC2
+              metric_name: CPUUtilization
+              statistic: average
+              unit: percent
+              target: 50
+              cooldown: 120
+          do_not_update:
+            - image_id
+      register: result
+    - debug: var=result
+
+'''
+RETURN = '''
+---
+instances:
+    description: List of active elastigroup instances and their details.
+    returned: success
+    type: dict
+    sample: [
+         {
+            "spotInstanceRequestId": "sir-regs25zp",
+            "instanceId": "i-09640ad8678234c",
+            "instanceType": "m4.large",
+            "product": "Linux/UNIX",
+            "availabilityZone": "us-west-2b",
+            "privateIp": "180.0.2.244",
+            "createdAt": "2017-07-17T12:46:18.000Z",
+            "status": "fulfilled"
+        }
+    ]
+group_id:
+    description: Created / Updated group's ID.
+    returned: success
+    type: string
+    sample: "sig-12345"
+
+'''
+
+HAS_SPOTINST_SDK = False
+__metaclass__ = type
+
+import os
+import time
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    import spotinst
+    from spotinst import SpotinstClientException
+
+    HAS_SPOTINST_SDK = True
+
+except ImportError:
+    pass
+
+eni_fields = ['description',
```

Comment:
Since all of these appear to be constants, it seems to me that making them tuples instead of lists would be good to avoid accidentally modifying them.

Reply:
Done 👍